### PR TITLE
Simplified HTTP calls with basic auth

### PR DESCRIPTION
**Pull request by jglick**
Avoids the problem noted in #2438. A simple search for `getApiToken` turned up much simpler idioms in `HudsonPrivateSecurityRealmTest` and `BasicHeaderProcessorTest`.

@reviewbybees esp. @oleg-nenashev

### CODE REVIEW

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java [hunk 1]
```diff
@@ -155,96 +127,8 @@ private WebClient createClientForUser(final String username) throws Exception {
...
-        CredentialsProvider provider = new BasicCredentialsProvider();
-        provider.setCredentials(new AuthScope("localhost", AuthScope.ANY_PORT, AuthScope.ANY_REALM),
-                                new UsernamePasswordCredentials(username, token));
-        wc.setCredentialsProvider(provider);
-        wc.login(username);
+        wc.addRequestHeader("Authorization", "Basic " + Scrambler.scramble(username + ":" + token));
         return wc;
     }
 
-    private void configureWebConnection(final WebClient wc, final String username, final String token) throws IOException {
```
Comment (jglick) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: Apparently unused.

**Comment by oleg-nenashev** [comment]
I'd rather keep both old-style and new-style tests.
From what I see the new approach has a bit different login-less logic, so the implementations are not equivalent for Jenkins logic.

**Comment by jglick** [comment]
> the new approach has a bit different login-less logic

Can you expand on what you think the differences are, and why you chose to use this more complicated logic?

**Comment by jglick** [comment]
Ping @oleg-nenashev?

**Comment by abayer** [comment]
:bee:

**Comment by recena** [comment]
🐝  Amazing clean-up

**Comment by jglick** [comment]
Have not gotten any more information from @oleg-nenashev about his reservations.

**Comment by oleg-nenashev** [comment]
Sorry, missed the question

**Comment by oleg-nenashev** [comment]
I've forgotten my point as well, so let's just merge it

**Pull request merged**